tests/Test/Xrefcheck/IgnoringSpec.hs

```diff
@@ -0,0 +1,50 @@
+{- SPDX-FileCopyrightText: 2019 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+
+module Test.Xrefcheck.IgnoringSpec where
+
+import qualified Data.ByteString.Lazy as BSL
+import Test.Hspec (Spec, describe, it, shouldBe)
+
+import Xrefcheck.Core
+import Xrefcheck.Scanners.Markdown
+
+spec :: Spec
+spec = do
+    describe "Parsing failures" $ do
+        it "Check if parsing incorrect markdowns produce exceptions" $ do
+            areIncorrect <- mapM isIncorrectMD failPaths
+            any id areIncorrect `shouldBe` True
+    describe "\"ignore link\" mode" $ do
+        it "Check \"ignore link\" performance" $ do
+            fi <- getFI "tests/markdowns/ignore_link.md"
+            (getRefs fi) `shouldBe` ["team"]
+    describe "\"ignore paragraph\" mode" $ do
+        it "Check \"ignore paragraph\" performance" $ do
+            fi <- getFI "tests/markdowns/ignore_paragraph.md"
+            (getRefs fi) `shouldBe` ["blog", "contacts"]
+    describe "\"ignore file\" mode" $ do
+        it "Check \"ignore file\" performance" $ do
+            fi <- getFI "tests/markdowns/ignore_file.md"
+            (getRefs fi) `shouldBe` []
+    where
+        failPaths =
+            [ "tests/markdowns/no_link.md"
+            , "tests/markdowns/no_paragraph.md"
+            , "tests/markdowns/unexpected_ignore_file.md"
+            , "tests/markdowns/unrecognised_option.md"
+            ]
+        
+        defFI = FileInfo [] []
+
+        parse path = parseFileInfo . decodeUtf8 <$> BSL.readFile path
+        
+        getFI path = fromRight defFI <$> parse path
+        
+        getRefs fi = map rName $ fi ^. fiReferences 
+
+        isIncorrectMD path = do
+            errOrInfo <- parse path
+            return $ isLeft errOrInfo
```

Comment:
:+1:

Reply:
I would like to see the following test: if a file contains "ignore file" command, this does not affect the list of anchors found in it (so if another file references an anchor in ignored file, this still works). 

I now understand that probably my initial suggestion to just erase the entire file on "ignore file" seems wrong in this regard. Indeed, it's debatable whether do we want to exclude file entirely or just skip checking links in it - I think ideally we could support both somehow, but if not, I think it's better to have the latter behavior - a file marked as ignored should still be referenced outside.

(though first let's complete the discussion about the necessity of `exclude file` just in case)

Reply:
Well okay, as we discussed separately, with such an approach `ignore paragraph`  would also have to be updated. That's all quite tricky to do and motivation for this is unclear. We will create a separate ticket for implementing commands in `notScanned` way.